etc/bash_completion.d/singularity.in

```diff
@@ -79,8 +85,16 @@ _singularity() {
                 COMPREPLY=( $(compgen -f -- ${cur}) )
             elif [[ ${cur} == -* ]] ; then
                 COMPREPLY=( $(compgen -W "--help --bind --scratch --home --pwd --contain --containall --ipc --pid --user --workdir --writable" -- ${cur}) )
+            elif [[ ${cur} == docker://*:* ]] ; then
+                # By default, bash considers ':' as a word separator.  We must thus strip the prefix docker: from our responses.
+                repo_name=$(echo ${cur} | sed -e 's|docker://\([-a-zA-Z0-9._]*\):.*|\1|')
+                response_var=$(echo ${cur} | sed -e 's|docker://[-a-zA-Z0-9._]*:||')
+                all_tags=$(PYTHONPATH="${exec_prefix}/libexec/singularity/python" python -c 'for i in __import__("docker.api").api.get_tags(namespace="library", repo_name=__import__("sys").argv[1]): print "%s" % i,' ${repo_name} 2>/dev/null)
```

Comment:
If this piece of voodoo doesn't make you frustrated with bash programming, I don't know what will!